server/ladder_service.py

```diff
@@ -402,9 +448,52 @@ async def shutdown(self):
             queue.shutdown()
 
 
+def game_name(*teams: List[Player]) -> str:
+    """
+    Generate a game name based on the players.
+    """
+
+    return " Vs ".join(_team_name(team) for team in teams)
+
+
+def _team_name(team: List[Player]) -> str:
+    """
+    Generate a team name based on the players. If all players are in the
+    same clan, use their clan tag, otherwise use the name of the first
+    player.
+    """
+    assert team
+
+    player_1_name = team[0].login
+
+    if len(team) == 1:
+        return player_1_name
+
+    clans = {player.clan for player in team}
+
+    if len(clans) == 1:
+        name = clans.pop() or player_1_name
+    else:
+        name = player_1_name
+
+    return f"Team {name}"
+
+
 def newbie_adjusted_ladder_mean(player: Player):
     """Get ladder rating mean with new player's always returning a mean of 0"""
     if player.game_count[RatingType.LADDER_1V1] > config.NEWBIE_MIN_GAMES:
         return player.ratings[RatingType.LADDER_1V1][0]
     else:
         return 0
+
+
+def alternate(iter1, iter2):
```

Comment:
If it ever needs to be used somewhere else then I think it would be worth extracting. For now I think it's OK. It is still pretty specific to the use case within this class.